projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php

```diff
@@ -467,18 +467,12 @@ public static function user_can_edit() {
 	 * @return bool Whether the post can be viewed
 	 */
 	public static function user_can_view_post() {
-		$user_id = get_current_user_id();
-		$post_id = get_the_ID();
-
-		if ( empty( $post_id ) ) {
-			// We are not viewing a post.
-			$post_id = 0;
-		}
-
+		$user_id   = get_current_user_id();
+		$post_id   = get_the_ID();
 		$cache_key = sprintf( '%d_%d', $user_id, $post_id );
 
-		if ( isset( self::$user_can_view_post_cache[ $cache_key ] ) ) {
-			return self::$user_can_view_post_cache[ $cache_key ];
+		if ( false === $post_id ) {
+			return true;
```

Comment:
```suggestion
			$post_id = 0;
```
I contemplated returning `true` directly here working on #31249.
I finally decided to return 0 in order to keep this `user_can_view_post()` as close as a soft cache layer and not introduce any "business" logic here.
Writing this, I am having double thoughts as right now the logic comes from get_**post**_access_level() right after, which doesn't feel right in a **page** context. Maybe we can rename that later when/if we tackle pages at some point.
I guess I would keep `0` as it as here..

Reply:
Fixed [508ae34](https://github.com/Automattic/jetpack/pull/31254/commits/508ae34b6fa58966abbc7d1d001838d7fe80f98d)